src/runners/javascript.js

```diff
@@ -41,26 +47,38 @@ function createWorker(dispatch) {
 // example run function
 const runJavaScript = (code = '', dispatch) => {
   const worker = createWorker(dispatch);
-  worker.postMessage(code);
+  const job = { command: RUN_CODE, data: code };
+  worker.postMessage(job);
 };
 
 // Header code is intended to be used for importing libraries, initialize screens, and so on.
 // Footer code is intended to be used for display runnning status, free resources, and so on.
-const runJavaScriptWithHeaderAndFooter = (
-  headerCode = '',
-  code = '',
-  footerCode = '',
-  dispatch
-) => {
+const runJavaScriptWithHeaderAndFooter = (config, dispatch) => {
+  const { headerCode, footerCode, input, data: code } = config;
   const worker = createWorker(dispatch);
+  dispatch({
```

Comment:
Yes. This dispatch need to be here because we need to establish a communication channel between frontend(ReactTerminal, etc) and worker before running header code.